examples/compact.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2018 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * Example program that compacts a set of indexes, and then uncompacts
+ * to the original set.
+ */
+
+#include <assert.h>
+#include <h3/h3api.h>
+#include <inttypes.h>
+#include <stdio.h>
+
+int main(int argc, char* argv[]) {
+    H3Index input[] = {
+        // All with the same parent index
+        0x8a2a1072b587fffL, 0x8a2a1072b5b7fffL, 0x8a2a1072b597fffL,
+        0x8a2a1072b59ffffL, 0x8a2a1072b58ffffL, 0x8a2a1072b5affffL,
+        0x8a2a1072b5a7fffL,
+        // These don't have the same parent indexas above.
+        0x8a2a1070c96ffffL, 0x8a2a1072b4b7fffL, 0x8a2a1072b4a7fffL};
+    int inputSize = sizeof(input) / sizeof(H3Index);
+    printf("Starting with %d indexes.\n", inputSize);
+
+    H3Index* compacted = calloc(inputSize, sizeof(H3Index));
+    int err = compact(input, compacted, inputSize);
+    // An error case can occur on e.g. duplicate input.
+    assert(err == 0);
+
+    int compactedCount = 0;
+    printf("Compacted:\n");
+    for (int i = 0; i < inputSize; i++) {
+        if (compacted[i] != 0) {
+            printf("%" PRIx64 "\n", compacted[i]);
+            compactedCount++;
+        }
+    }
+    printf("Compacted to %d indexes.\n", compactedCount);
+
+    int uncompactRes = 10;
+    int uncompactedSize = maxUncompactSize(compacted, inputSize, uncompactRes);
+    H3Index* uncompacted = calloc(uncompactedSize, sizeof(H3Index));
```

Comment:
Hate to nitpick but you could use stack allocation here.

Reply:
I originally didn't use stack allocation because that would require including not only the compiler detection code (`../cmake/CheckAlloca.cmake` and `../cmake/CheckVLA.cmake`) but also including the internal header `stackAlloc.h`.

I could refactor the stack allocation code to be self contained in its own directory, but this seems like it produces a simple and easily portable set of examples.

Reply:
Oh right. I forgot that you weren't relying on the existing CMake includes in the main CMake file. No need to change.